routemaster/client/assertion_helpers.rb

```diff
@@ -0,0 +1,18 @@
+require 'uri'
+
+module Routemaster
+  class Client
+    module AssertionHelpers
+      def assert_valid_url_throwing_error!(url, error_class)
+        begin
+          uri = URI.parse(url)
+          unless uri.is_a? URI::HTTPS
+            raise error_class, "url '#{url}' is invalid, must be an https url"
```

Comment:
Well yeah a valid url of a different scheme passes the URI regex, but it fails our validation of it being an https url. Do you have a better error message in mind?